### DISCUSSION TITLE
Suggestion: avoid using two or more times the same color when calling `quickdraw`

### DISCUSSION THREAD

**Discussion by mthh** [neocarto/bertin#77] [Ideas]
When calling `quickdraw` with an array of layers (e.g. 3), sometimes several layers have the same color (e.g. 2 out of 3).

Maybe when calling `quickdraw` you could compute the color to use in `quickdraw` (to avoid having the same color used several times) rather than leaving this task to the `simple` function (https://github.com/neocarto/bertin/blob/main/src/layers/simple.js#L66) ?
(at least when a reasonable number of layers, e.g. 8 or less, is used)

When trying to reproduce it in your [@neocartocnrs/bertin-js-quickdraw-table2geo](https://observablehq.com/@neocartocnrs/bertin-js-quickdraw-table2geo) notebook I got 3 times the same color :

![quickdraw](https://user-images.githubusercontent.com/12172162/197230493-804b3865-3e6e-474a-8b1a-22a5abf3344f.png)

**Comment by neocarto** [comment]
Not so easy because quickdraw is a wrapper around draw. 
Well. I'll think about it. 
Thanks Matt.

**Comment by mthh** [comment]
Not a PR but I was thinking something like the following would work :
(as the `fill` property is passed unmodified to `draw` and then to `simple` ?)

https://github.com/neocarto/bertin/blob/main/src/quickdraw.js#L4 :
```js
if (Array.isArray(layers)) {
  // The original color array
  let cols = ["#66c2a5", "#fc8d62", "#8da0cb", "#e78ac3", "#a6d854", "#ffd92f", "#e5c494", "#b3b3b3"];

  // Shuffle it so that succesive calls to quickdraw don't return them in same order
  cols.sort(() => (Math.random() > 0.5) ? 1 : -1)

  let myarray = layers.map((x, i) =>({
    type: "layer",
    geojson: x,
    fill: cols[i % cols.length], // Use them in their new order, with repetition if more than cols.length layers
  }));

  return draw({ params: { margin: m, width: w }, layers: myarray });
} else {
  ...
```

Well, this is a very low priority/low importance concern ^^
